draft-ietf-quic-recovery.md

```diff
@@ -1067,14 +1066,18 @@ Pseudocode for OnLossDetectionTimeout follows:
 
 ~~~
 OnLossDetectionTimeout():
-  if (crypto packets are in flight):
+  packets_lost = false
+  for pn_space in [ Initial, Handshake, ApplicatonData ]:
+    if (loss_times[pn_space] != 0):
+      // Time threshold loss Detection
+      DetectLostPackets(pn_space)
+      packets_lost = true
+  // Don't retransmit all crypto data if a packet was just lost.
+  if (!packets_lost &&
```

Comment:
If `packets_lost = true`, we'd send PTO packets. I don't think this is the intention. We should return early in that case.

Reply:
I added GetEarliestLossTime, which I think makes this more efficient and correct.